docs/template_plugin/tests/functional/op_reference/logical_and.cpp

```diff
@@ -0,0 +1,101 @@
+        result << "oType=" << param.outType;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const PartialShape& input_shape1, const PartialShape& input_shape2, const element::Type& input_type) {
+        const auto in = std::make_shared<op::Parameter>(input_type, input_shape1);
+        const auto in2 = std::make_shared<op::Parameter>(input_type, input_shape2);
+        const auto logical_and = std::make_shared<op::v1::LogicalAnd>(in, in2);
+        return std::make_shared<Function>(NodeVector {logical_and}, ParameterVector {in, in2});
+    }
+};
+
+TEST_P(ReferenceLogicalAndLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+template <element::Type_t IN_ET>
+std::vector<LogicalAndParams> generateLogicalAndParams(const ngraph::element::Type& type) {
+    using T = typename element_type_traits<IN_ET>::value_type;
+    std::vector<LogicalAndParams> logicalAndParams {
+        // 1D // 2D // 3D // 4D
+        LogicalAndParams(ngraph::PartialShape {2, 2}, ngraph::PartialShape {2, 2}, type, ngraph::element::boolean,
+                std::vector<T> {true, false, true, false},
+                std::vector<T> {false, true, true, false},
+                std::vector<char> {false, false, true, false}),
+        LogicalAndParams(ngraph::PartialShape {1}, ngraph::PartialShape {1},  type, ngraph::element::boolean,
+                std::vector<T> {true},
+                std::vector<T> {true},
+                std::vector<char> {true}),
+
+        LogicalAndParams(ngraph::PartialShape {2, 1, 2, 1}, ngraph::PartialShape {1, 1, 2, 1}, type, ngraph::element::boolean,
+                std::vector<T> {true, false, true, false},
+                std::vector<T> {true, false},
+                std::vector<char> {true, false, true, false}),
+        LogicalAndParams(ngraph::PartialShape {3, 4}, ngraph::PartialShape {3, 4}, type, ngraph::element::boolean,
+                std::vector<T> {true, true, true, true, true, false, true, false, false, true, true, true},
+                std::vector<T> {true, true, true, true, true, false, true, false, false, true, true, false},
+                std::vector<char> {true, true, true, true, true, false, true, false, false, true, true, false})};
+    return logicalAndParams;
+}
+
+std::vector<LogicalAndParams> generateLogicalAndCombinedParams() {
+    const std::vector<std::vector<LogicalAndParams>> logicalAndTypeParams {generateLogicalAndParams<element::Type_t::u8>(ngraph::element::boolean)};
+    std::vector<LogicalAndParams> combinedParams;
+    std::for_each(logicalAndTypeParams.begin(), logicalAndTypeParams.end(), [&](std::vector<LogicalAndParams> params) {
+        combinedParams.insert(combinedParams.end(), params.begin(), params.end());
+    });
+    return combinedParams;
+}
+
+INSTANTIATE_TEST_SUITE_P(smoke_LogicalAnd_With_Hardcoded_Refs, ReferenceLogicalAndLayerTest, ::testing::ValuesIn(generateLogicalAndCombinedParams()),
```

Comment:
you don't need anymore logical_and_in.cpp backend test when one for template plugin is created.

Reply:
Right, removed